ReactWindows/ReactNative/Views/View/ReactViewManager.cs

```diff
@@ -0,0 +1,270 @@
+
+                        if (file != null)
+                        {
+                            var props = await file.GetBasicPropertiesAsync();
+                            var type = file.ContentType;
+                            var path = drop ? "urn:future-access-list:" + Uri.EscapeUriString(GetAccessToken(file)) : "";
+
+                            files.Add(new JObject
+                            {
+                                { "name", file.Name },
+                                { "size", props.Size },
+                                { "type", type },
+                                { "uri", path }
+                            });
+
+                            items.Add(new JObject
+                            {
+                                { "kind", "file" },
+                                { "type", type }
+                            });
+
+                            types.Add(type);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        System.Diagnostics.Debug.WriteLine("GetDataTransferInfo: " + ex);
+                    }
+                }
+            }
+
+            return new JObject
+            {
+                { "files", files },
+                { "items", items },
+                { "types", types }
+            };
+        }
+
+        private static string GetAccessToken(StorageFile file)
+        {
+            var futureAccessList = StorageApplicationPermissions.FutureAccessList;
+            var existingEntries = futureAccessList.Entries;
+            var capacity = futureAccessList.MaximumItemsAllowed;
+            var now = DateTime.Now;
+
+            // Use more aggresive timeout when number of items exceed 100 or half of the capacity
+            // Note: 100 is an arbitary limit in case that OS reduces the capacity (it is 1000 in RS1). 
+            var expiry = existingEntries.Count >= capacity / 2 || existingEntries.Count >= 100 ?
+                now.AddDays(-1) :
+                now.AddDays(-7);
+
+            foreach (var entry in existingEntries)
+            {
+                if (DateTime.TryParse(entry.Metadata, out DateTime time) && time < expiry)
+                {
+                    futureAccessList.Remove(entry.Token);
+                }
+            }
+
+            return futureAccessList.Add(file, now.ToString(System.Globalization.CultureInfo.InvariantCulture));
+        }
+
+        private static async void OnDragEnter(object sender, DragEventArgs args)
+        {
+            var view = sender as BorderedCanvas;
+            var data = await GetDataTransferInfo(args.DataView);
+
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(new DragDropEvent(view.GetTag(), "topDragEnter", data));
+        }
+
+        private static async void OnDragOver(object sender, DragEventArgs args)
```

Comment:
>OnDragOver [](start = 34, length = 10)

Are we going to expose any of the `DragUIOverride` properties?

Reply:
Er, I guess some of that could be handled by JavaScript.

---
In reply to: [159503898](https://github.com/Microsoft/react-native-windows/pull/1549#discussion_r159503898) [](ancestors = 159503898)

Reply:
We can't expose `DragUIOverrides` easily. These properties correspond to `effectAllowed` and `dropEffect` in the JS [drag event](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer) and the pattern JS apps use is to set these props in the drag over handler:

```js
element.onDragOver = event => {
  if (_that_ && _this_) {
    event.effectAllowed = "copy";
    event.dropEffect = "copy";
  }
};
```

We can't do this without a good amount of hackery and since S4L didn't need this, I implemented the simplest option: `effectAllowed=copy` and `dropEffect=copy` are implied if `allowDrop=true`.